pkg/cloudprovider/provider/aws/provider.go

```diff
@@ -509,6 +514,7 @@ func (p *provider) Create(machine *v1alpha1.Machine, data *cloudprovidertypes.Pr
 					DeleteOnTermination: aws.Bool(true),
 					VolumeType:          aws.String(config.DiskType),
 					Iops:                config.DiskIops,
+					Encrypted:           pointer.BoolPtr(config.EBSVolumeEncrypted),
```

Comment:
Can we add an e2e test where this is enabled?

Reply:
I ran the current e2e tests with this sat to `true` and the tests are passed( I am not sure though if this change has any influence on the machine, e.g: the machine will not be provisioned). We can add another test with this enabled. WDYT?

Reply:
sgtm, but I don't think we need to test this on all kube versions and all distributions, one version on one distribution should be enough